src/components/TodoComponent/TodoComponent.tsx

```diff
@@ -0,0 +1,190 @@
+
+  useEffect(() => inputRef.current?.focus(), [editing]);
+
+  const { upatingTodos, addTodoForUpdate, removeTodoForUpdate }
+    = useContext(TodosContext);
+
+  const deleting = upatingTodos.some(({ id }) => todo.id === id);
+
+  const showUpdating = deleting || loading || updating;
+
+  function removeTodo(): void {
+    addTodoForUpdate(todo);
+
+    deleteTodo(todo.id)
+      .then(() => onDelete(todo.id))
+      .catch(() => {
+        onError('Unable to delete a todo');
+        removeTodoForUpdate(todo);
+      })
+      .finally(() => removeTodoForUpdate(todo));
+  }
+
+  const toogleCheck = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const changedTodo: Partial<Todo> = {
+      id: todo.id,
+      completed: event.target.checked,
+    };
+
+    setUpdating(true);
+    changeTodo(changedTodo)
+      .then(res => {
+        onUpdate(res);
+      })
+      .catch(() => onError('Unable to update a todo'))
+      .finally(() => setUpdating(false));
+  };
+
+  function handleDoubleClick(): void {
+    setEditing(true);
+  }
+
+  function cancelEditing():void {
+    setUpdating(false);
+    setEditing(false);
+  }
+
+  function saveEditedTodo() {
+    const value = newTitle.trim();
+
+    // eslint-disable-next-line no-extra-boolean-cast
+    if (!!value && value !== todo.title) {
+      setUpdating(true);
+
+      const updatedTodo: Partial<Todo> = {
+        id: todo.id,
+        title: value,
+      };
+
+      changeTodo(updatedTodo)
+        .then((editetTodo => {
+          onUpdate(editetTodo);
+          setEditing(false);
+        }))
+        .catch(() => onError('Unable to update a todo'))
+        .finally(() => setUpdating(false));
+    }
+
+    if (!value) {
+      removeTodo();
+    }
+
+    if (value === todo.title) {
+      cancelEditing();
+    }
+  }
+
+  function handleKey(event: React.KeyboardEvent<HTMLInputElement>) {
+    if (event.key === 'Escape') {
+      cancelEditing();
+    }
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    saveEditedTodo();
+  }
+
+  return (
+    <div data-cy="Todo" className={cn('todo', { completed: todo.completed })}>
+
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={toogleCheck}
+        />
+      </label>
+
+      {editing && (
+        <form onSubmit={handleSubmit}>
+          <input
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={newTitle}
+            ref={inputRef}
+            onBlur={saveEditedTodo}
+            onKeyDown={handleKey}
+            onChange={(e) => setNewTitle(e.target.value)}
+          />
+          <LoadOverlay showUpdating={showUpdating} />
+        </form>
+      )}
```

Comment:
```suggestion
      {!!editing && (
        <form onSubmit={handleSubmit}>
          <input
            data-cy="TodoTitleField"
            type="text"
            className="todo__title-field"
            placeholder="Empty todo will be deleted"
            value={newTitle}
            ref={inputRef}
            onBlur={saveEditedTodo}
            onKeyDown={handleKey}
            onChange={(e) => setNewTitle(e.target.value)}
          />
          <LoadOverlay showUpdating={showUpdating} />
        </form>
      )}
```
In render with && best way use convert to boolean